components/calendar/src/cal/chinese.rs

```diff
@@ -584,47 +583,37 @@ impl<R: Rules> DateFieldsResolver for LunarChinese<R> {
         month_code: types::ValidMonthCode,
         options: DateFromFieldsOptions,
     ) -> Result<u8, MonthCodeError> {
-        match year.parse_month_code(month_code) {
-            ComputedOrdinalMonth::Exact(val) => Ok(val),
-            ComputedOrdinalMonth::LeapToNormal(val) => {
-                if options.overflow == Some(Overflow::Constrain) {
-                    Ok(val)
-                } else {
-                    Err(MonthCodeError::NotInYear)
-                }
-            }
-            ComputedOrdinalMonth::NotFound => Err(MonthCodeError::NotInCalendar),
+        // 14 is a sentinel value, greater than all other months, for the purpose of computation only;
+        // it is impossible to actually have 14 months in a year.
+        let leap_month = year.leap_month().unwrap_or(14);
+
+        // leap_month is the ordinal of the second month, so subtract 1 to get the code
+        if month_code == ValidMonthCode::new_unchecked(leap_month - 1, true) {
+            return Ok(leap_month);
         }
+
+        let (number @ 1..13, leap) = month_code.to_tuple() else {
+            return Err(MonthCodeError::NotInCalendar);
+        };
+
+        if leap && options.overflow != Some(Overflow::Constrain) {
+            // wrong leap month and not constraining
+            return Err(MonthCodeError::NotInYear);
+        }
+
+        // add one if we're after the leap month if
```

Comment:
```suggestion
        // If our month is after the leap month, we need to add one to the
        // month number to get the ordinal month to account for the extra
        // leap month in between.
```

Doesn't have to be that verbose, but this is currently a partial sentence.